### PR TITLE
Wrong count with custom filter.

### PR DESCRIPTION
**Pull request by nedSaf**
#339

@amitaibu @mateu-aguilo-bosch 

I have added a custom filtering condition to the count as well.
selection_040

This solved the count issue for me.

### CODE REVIEW

File: plugins/restful/RestfulDataProviderEFQ.php [hunk 1]
```diff
@@ -186,6 +186,13 @@ public function getQueryCount() {
 
     $this->queryForListFilter($query);
 
+    // Add custom filters.
+    if($conditions = $this->getQueryForList()->fieldConditions) {
```
Comment (amitaibu) [inline plugins/restful/RestfulDataProviderEFQ.php:190]: this looks wrong - as I would expect `$this->queryForListFilter($query);` to already do the filtering.

**Comment by amitaibu** [comment]
(btw, I have a feeling it's related to the fact that the filter is done on a field which is a `resource`)

**Comment by amitaibu** [comment]
> (btw, I have a feeling it's related to the fact that the filter is done on a field which is a resource)

No, that's not the case.

**Comment by nedSaf** [comment]
@amitaibu 
If i ditch the `getQueryForList` custom filter and add a public field and filter through that one like:
`?filter[group]=7`, It works without any  modifications to the restful code and with the proper count.
You think i should change my approach ?

**Comment by nedSaf** [comment]
From what i saw in the code `$this->queryForListFilter($query);` Doesn't do custom filtering, Only filters according to public fields and a filter query param.

**Comment by amitaibu** [comment]
Seems the mistake is that you should use `?filter[group]=24`.

**Pull request closed without merging**